src/Microsoft.Sbom.Api/Workflows/SBOMRedactionWorkflow.cs

```diff
@@ -50,44 +50,61 @@ public virtual async Task<bool> RunAsync()
         var sboms = await GetValidSbomsAsync();
 
         log.Information($"Running redaction on the following SBOMs: {string.Join(' ', sboms.Select(sbom => sbom.Key))}");
-        foreach (var (sbomPath, validatedSbom) in sboms)
+
+        try
         {
-            try
+            foreach (var (sbomPath, validatedSbom) in sboms)
             {
-                var outputPath = GetOutputPath(sbomPath);
-                var redactedSpdx = await this.sbomRedactor.RedactSBOMAsync(validatedSbom);
-                using (var outStream = fileSystemUtils.OpenWrite(outputPath))
+                try
```

Comment:
Nested `try` blocks in the same method are something of a code smell. Would it make sense to refactor lines 58-72 into a private method, then just call that method from inside the loop?

Reply:
Yep, it looks like you might be looking at an older commit, I didn't like this either so I refactored in a subsequent commit.